golang/UPDATING.md

```diff
@@ -4,32 +4,21 @@
 
 #### One-Time Setup
 
-1. Download and setup Protocol Buffer release from https://github.com/protocolbuffers/protobuf/releases (if you haven't already done this for another language).  As of February 2019 we're using v3.7 release, which is compatible with proto2 .proto files.
-1. Download [Go](https://golang.org/dl/)
+1. Download and install [Docker](https://docs.docker.com/get-docker/)
 
 #### Every time `gtfs-realtime.proto` changes
 
-1. Regenerate the language binding source from gtfs-realtime.proto by running the following from the `golang` directory:
+1. Run the following from the project root folder:
 
     ```
-    go get -u github.com/golang/protobuf/protoc-gen-go
-    protoc --go_out=./gtfs/ --proto_path=.. ../gtfs-realtime.proto
+    docker build -t gtfs-golang -f golang/Dockerfile .
+    # -it to make sure docker run can be killed with ctrl-c
+    # -t uses TTY, which causes linux to include carriage returns, which are stripped using tr
+    docker run -it --rm gtfs-golang cat /lib/gtfs/gtfs-realtime.pb.go | tr -d '\r' > golang/gtfs/gtfs-realtime.pb.go
     ```
 
 1. Add the license header back to the generated source file.
 
-1. Change the line:
-
-    ```golang
-    package transit_realtime
-    ```
-
-    to:
-
-    ```golang
-    package gtfs
-    ```
-
 1. Test the generated code from the `golang/gtfs` directory:
```

Comment:
The tests still require having go installed on your local workstation, per the removed "Download Go" instructions above.  Is there any way to integrate the test into the Docker build process?

Reply:
@bdferris-v2 Great suggestion! I just pushed up a change to include the unit testing run inside the docker build. I also validated that when I cause the unit test to fail, the docker build process also fails:
```
| $ docker build -t gtfs-golang -f golang/Dockerfile .
[+] Building 14.2s (17/17) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                       0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                            0.0s
 => [internal] load metadata for docker.io/library/golang:1.19.3                                                                                                                                           0.2s
 => [ 1/13] FROM docker.io/library/golang:1.19.3@sha256:10e3c0f39f8e237baa5b66c5295c578cac42a99536cc9333d8505324a82407d9                                                                                   0.0s
 => => resolve docker.io/library/golang:1.19.3@sha256:10e3c0f39f8e237baa5b66c5295c578cac42a99536cc9333d8505324a82407d9                                                                                     0.0s
 => [internal] load build context                                                                                                                                                                          0.0s
 => => transferring context: 1.38kB                                                                                                                                                                        0.0s
 => CACHED [ 2/13] WORKDIR /lib                                                                                                                                                                            0.0s
 => CACHED [ 3/13] COPY golang/gtfs/go.mod /lib/go.mod                                                                                                                                                     0.0s
 => CACHED [ 4/13] COPY golang/gtfs/go.sum /lib/go.sum                                                                                                                                                     0.0s
 => CACHED [ 5/13] COPY gtfs-realtime.proto /gtfs-realtime.proto                                                                                                                                           0.0s
 => CACHED [ 6/13] RUN mkdir gtfs                                                                                                                                                                          0.0s
 => [ 7/13] COPY golang/gtfs/gtfs_test.go /lib/gtfs/gtfs_test.go                                                                                                                                           0.0s
 => [ 8/13] COPY golang/gtfs/vehicle_position.pb /lib/gtfs/vehicle_position.pb                                                                                                                             0.0s
 => [ 9/13] RUN apt-get update && apt-get install -y unzip                                                                                                                                                 3.7s
 => [10/13] RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v21.9/protoc-21.9-linux-x86_64.zip     && unzip -o protoc-21.9-linux-x86_64.zip -d /usr/local bin/protoc     && un  0.8s
 => [11/13] RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28.1                                                                                                                            4.9s
 => [12/13] RUN protoc --go_out=./ --go_opt=Mgtfs-realtime.proto=./gtfs --proto_path=.. gtfs-realtime.proto                                                                                                0.3s
 => ERROR [13/13] RUN go test ./...                                                                                                                                                                        4.0s
------
 > [13/13] RUN go test ./...:
#17 0.254 go: downloading google.golang.org/protobuf v1.26.0
#17 3.877 --- FAIL: TestFeedMessage (0.00s)
#17 3.877     gtfs_test.go:29: Expected entity length: 2, got: 1
#17 3.877 FAIL
#17 3.877 FAIL	github.com/MobilityData/gtfs-realtime-bindings/golang/gtfs/gtfs	0.003s
#17 3.877 FAIL
------
executor failed running [/bin/sh -c go test ./...]: exit code: 1
```